third-party/rsutils/include/rsutils/concurrency/event.h

```diff
@@ -78,9 +81,7 @@ class event
     bool wait( std::chrono::duration< Rep, Period > const & timeout ) const
     {
         std::unique_lock< std::mutex > lock( _m );
-        if( ! _is_set )
-            _is_set = ( std::cv_status::timeout != _cv.wait( lock, timeout ) );
-        return _is_set;
+        return _is_set || std::cv_status::timeout != _cv.wait_for( lock, timeout );
```

Comment:
Do you think we should use a predicate to protect against "sporadic wakeups"? It is listed as a possibility in the documentation, but I don't know if it's a real issue

Reply:
Well, the Python code did the `_is_set = ` here which is why we had it. My guess is that this can happen when someone else gets the mutex and clears it before you get it? But then it really depends on the semantics: you did wake up because someone set the event...